laurent22/joplin
# Desktop: Resolves #7880: Paste as Text only working on hotkeys on Windows

Resolves https://github.com/laurent22/joplin/issues/7880

When making the implementation for the Paste as Text functionality I use the `clipboard.read` function which is still experimental and doesn't seem to work on Windows.

To fix that I just use the simpler function `readText`

https://www.electronjs.org/docs/latest/api/clipboard

## How to Test

- On Windows
- Copy content to the clipboard
- Inside the Rich Text Editor, right-click inside the editor and select 'Paste as Text' or select the option from the top menu Edit -> Paste as Text
- The copied content should appear on the screen

If the command works with other shortcuts:
- On Windows 
- Rebind 'Paste as Text' to Ctrl+M
- Copy content to Clipboard
- Inside the Rich Text Editor, try to paste content with the shortcut
- The copied content should appear on the screen

With Google Docs:
- On Windows, inside the Rich Text Editor
- Copying content from Google Docs (eg: https://docs.google.com/document/d/1NUVcHRDdRDb1jUBpedIPSHGuu0dbJM8ppiLt1LMJR8M/edit?usp=sharing) 
- Pasting the content with Paste as Text should insert plain text as opposed to Paste which shows content formatted as the original document

With empty content:
- On Windows, inside the Rich Text Editor
- Clean clipboard (Windows+V to be able to access clipboard utility)
- Paste as Text does not appear as an option inside the Context Menu since it has nothing to paste
- Pasting with the upper menu option should call the command, but nothing should be inserted since there is no content inside the clipboard
